src/api/providers/sapaicore.ts

```diff
@@ -361,11 +391,15 @@ export class SapAiCoreHandler implements ApiHandler {
 							if (data.metadata?.usage) {
 								const inputTokens = data.metadata.usage.inputTokens || 0
 								const outputTokens = data.metadata.usage.outputTokens || 0
+								const cacheReadInputTokens = data.metadata.usage.cacheReadInputTokens || 0
```

Comment:
From my testing, cache usage information is not returned when using `converse-stream`. Despite that, it is implicitly reflected in `usage.inputTokens` showing low token count (due to cache)

Reply:
According to the [documentation](https://docs.aws.amazon.com/bedrock/latest/userguide/conversation-inference-call.html) it seems only converse (non-stream) returns such information

Reply:
Yes, cache mechanism does work

Here is an example of a raw event:
```
:message-typeevent{"metrics":{"latencyMs":17329},"p":"abcdefghijklmnopqrstuvwxyzAB","usage":{"cacheReadInputTokenCount":1490,"cacheReadInputTokens":1490,"cacheWriteInputTokenCount":0,"cacheWriteInputTokens":0,"inputTokens":4,"outputTokens":832,"totalTokens":2326}}%�qI
```

total_tokens = 2326
output_tokens = 832
input_tokens = 4
cache_read_input_tokens = 2326-832-4= 1490


The calculation logic may be adjusted according to the example calculation above

Reply:
Agree. 

The equation shall be :
totalTokens = cacheReadInputTokenCount + cacheWriteInputTokenCount + inputTokens + outputTokens

I will close this PR as another PR for the same purpose is merged.

I am creating another PR to calibrate the inputToken, otherwise the tokens number in context window is incorrect, context window might show a few token used but it could already exceeds the context window limit.